3rdparty/exported/quickjs/quickjs.c

```diff
@@ -48191,7 +48198,12 @@ static JSValue get_date_string(JSContext *ctx, JSValueConst this_val,
 }
 
 /* OS dependent: return the UTC time in ms since 1970. */
-static int64_t date_now(void) {
+// vvv CCF Patch vvv
+// Implement date_now as a macro, to additionally capture ctx argument from
+// callsite
+#define date_now() _date_now(ctx)
+static int64_t _date_now(JSContext* ctx) {
+// ^^^ CCF Patch ^^^
```

Comment:
An extremely unrelated aside: No sensible meta-programming/code generation language would let us do this! So maybe dumb old text-replacement macros still have a place.